blueocean-dashboard/src/main/js/components/Capability.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react';
+import update from 'react-addons-update';
+import { classMetadataStore } from '@jenkins-cd/js-extensions';
+
+export const capabilityStore = classes => ComposedComponent => class extends Component {
```

Comment:
In fact, maybe a general link to some docs explaining the composition pattern (Vs mixins e.g. https://facebook.github.io/react/blog/2016/07/13/mixins-considered-harmful.html).